src/database/models/hook.model.ts

```diff
@@ -0,0 +1,41 @@
+import { CreationOptional, DataTypes, ForeignKey, InferAttributes, InferCreationAttributes, Model, ModelStatic, Sequelize } from "sequelize";
+import { sql } from '@sequelize/core';
+import sequelize from "..";
+
+interface IHook extends Model<InferAttributes<IHook>, InferCreationAttributes<IHook>> {
+    id: CreationOptional<number>,
+    position: number[];
+    current_panel_id: ForeignKey<number>,
+    next_panel_set_id: ForeignKey<number>
+}
+
+let Hook: ModelStatic<IHook>;
+
+const define = (sequelize: Sequelize): void => {
+    Hook = sequelize.define<IHook>(
+        'hook',
+        {
+            id: {
+                type: DataTypes.INTEGER,
+                primaryKey: true,
+                autoIncrement: true,
+                allowNull: false
+            },
+            position: {
+                type: DataTypes.ARRAY(DataTypes.FLOAT),
+                allowNull: false
+            },
+            current_panel_id: {
+                type: DataTypes.INTEGER,
+                allowNull: false
+            },
+            next_panel_set_id: {
+                type: DataTypes.INTEGER,
+                allowNull: true,
+                unique: true
+            }
+        }
+    );
+}
```

Comment:
```js
PanelSet.hasMany(Hook, {
  foreignKey: 'current_panel_id',
  as: 'hook_current_panel_set'
});
PanelSet.hasOne(Hook, {
  foreignKey: 'next_panel_set_id',
  as: 'panel_set_origin_hook'
});
```
We need foreign keys somewhere?

Reply:
We were going to wait to do associations until we have all of our models pushed into main (I believe)